src/cryptonote_protocol/quorumnet.cpp

```diff
@@ -216,12 +230,20 @@ class peer_info {
                     "tcp://" + epee::string_tools::get_ip_string_from_int32(proof.public_ip) + ":" + std::to_string(proof.quorumnet_port)));
             });
 
-        compute_peers(qbegin, qend, opportunistic);
+        compute_validator_peers(qbegin, qend, opportunistic);
+
+        if (include_workers) {
+            for (auto qit = qbegin; qit != qend; ++qit) {
+                auto &w = (*qit)->workers;
+                for (size_t i = 0; i < w.size(); i++)
+                    add_peer(w[i]);
+            }
```

Comment:
This looks unnecessary -- the loop added just above looks like it is doing exactly the same thing (except that it also handles `exclude`).